src/reducers-actions/trades.ts

```diff
@@ -14,39 +23,55 @@ interface WithReverts {
   reverts: TradeReversion[]
 }
 
-type OverwriteTradesActionType = Actions<'OVERWRITE_TRADES', Omit<TradesState, 'pendingTrades'> & WithReverts>
-type AppendTradesActionType = Actions<'APPEND_TRADES', Required<Omit<TradesState, 'pendingTrades'>> & WithReverts>
-type UpdateBlockActionType = Actions<'UPDATE_BLOCK', Required<Pick<TradesState, 'lastCheckedBlock'>>>
-type ReducerActionType = Actions<ActionTypes, TradesState & WithReverts>
+interface WithNetworkId {
+  networkId: number
+}
 
-interface Params {
+// ******** ACTION TYPES
+
+type OverwriteTradesActionType = Actions<
+  'OVERWRITE_TRADES',
+  Omit<TradesStateSingleNetwork, 'pendingTrades'> & WithReverts & WithNetworkId
+>
+type AppendTradesActionType = Actions<
+  'APPEND_TRADES',
+  Required<Omit<TradesStateSingleNetwork, 'pendingTrades'>> & WithReverts & WithNetworkId
+>
+type UpdateBlockActionType = Actions<
+  'UPDATE_BLOCK',
+  Required<Pick<TradesStateSingleNetwork, 'lastCheckedBlock'>> & WithNetworkId
+>
+type ReducerActionType = Actions<ActionTypes, TradesStateSingleNetwork & WithReverts & WithNetworkId>
+
+interface Params extends WithNetworkId {
   trades: Trade[]
   reverts: TradeReversion[]
   lastCheckedBlock?: number
 }
 
-export const overwriteTrades = ({ trades, reverts, lastCheckedBlock }: Params): OverwriteTradesActionType => ({
+// ******** REDUCER FUNCTIONS
+
+export const overwriteTrades = (params: Params): OverwriteTradesActionType => ({
   type: 'OVERWRITE_TRADES',
-  payload: { trades, reverts, lastCheckedBlock },
+  payload: params,
 })
 
-export const appendTrades = ({ trades, reverts, lastCheckedBlock }: Required<Params>): AppendTradesActionType => ({
+export const appendTrades = (params: Required<Params>): AppendTradesActionType => ({
   type: 'APPEND_TRADES',
-  payload: { trades, reverts, lastCheckedBlock },
+  payload: params,
 })
 
-export const updateLastCheckedBlock = (lastCheckedBlock: number): UpdateBlockActionType => ({
+export const updateLastCheckedBlock = (lastCheckedBlock: number, networkId: number): UpdateBlockActionType => ({
   type: 'UPDATE_BLOCK',
-  payload: { lastCheckedBlock },
+  payload: { lastCheckedBlock, networkId },
 })
 
-// TODO: store to/load from localStorage
-export const INITIAL_TRADES_STATE = { trades: [], pendingTrades: new Map<string, Trade[]>() }
-
 function buildTradeRevertKey(batchId: number, orderId: string): string {
   return batchId + '|' + orderId
 }
 
+// ******** HELPERS
+
 function groupByRevertKey<T extends EventWithBlockInfo>(list: T[], initial?: Map<string, T[]>): Map<string, T[]> {
```

Comment:
idk why buy this function just screams Leandro, and not in any bad way, just is maybe sth I've seen you handling/man-handling/coding out for this app a alot in other PRs. I have these kinds for Anxo and Dima, now I found your fav type! ;)

Reply:
Wait, what type? Now I'm curious. Do I do that a lot? :thinking: